### ISSUE TITLE
[bug] fresh install of gdrive incorrectly deletes and modifies files on first push

### ISSUE THREAD

**Issue by barteend** [odeke-em/drive#160]
I just installed gdrive on Kubuntu Trusty 14.04 LTS.

On my first drive push command, the resolve shows a long long list of files that are going to be deleted, modified or added. The added files have the same path and filename as files that are going to be deleted.
To give you an impression:
```
Deletion count 457 dest: 481.68MB
Modification count 24 src: 7.29MB dest: 5.89MB
Addition count 15 src: 568.87KB
```

The files that are going to be deleted don't exist on my local filesystem.
Probably they were not downloaded to my filesystem during the first pull.

On the question `Proceed with the changes? [Y/n]` I obviously typed 'n' ;-)

Then I tried a drive pull, just to see what happens. This is what I got:
```
drive pull
Resolving...
md5Checksum: `20141105_150111.mp4` (452.74MB)
might take time to checksum.
These 1 file(s) would be overwritten. Use -ignore-conflict to override this behaviour
/A/A/a.ott 
conflicts have prevented a pull operation
```

Note: I changed the filename in this post to /A/A/a.ott.

The file /A/A/a.ott is one of the files that was on the deletion list of the drive push command.

If there is any more information you need to fix this bug, please let me know.

Can anybody tell me how I can solve this problem? So I can start using gdrive as it is supposed to work :-)

P.S. Grive is also still installed on my system. I don't think this should cause the problem, but I just want to provide all the information that might be important.

**Comment by odeke-em** [comment]
Hello there @barteend! Thank you for reporting this and welcome to drive.

So whenever you try to pull content from a directory that isn't available on one remote, the goal is to mirror the source on the destination. This entails deleting content that doesn't exist locally or modifying or adding. I'd advise you if you haven't already to take a look at the README.md file as you'll see how to use drive to avoid such situations, namely you might have clashing filenames within the same directory (your file system cannot handle this). Also you can skip expensive checksum-ing by using flag `--ignore-checksum`, as well as pulling specific files and folders instead of the whole thing.

Any other program on your system should not cause any problem.
Thank you for providing as much information.

Please let me know if this makes sense.

**Comment by odeke-em** [comment]
Am marking this as works as intended. Please don't hesitate to let me know if there is any problem.

**Comment by barteend** [comment]
Thank you for your fast reply @odeke-em ! :-) :+1: 

I've read the complete README.md

I understand that when I do a pull, then content (files and directories) that don't exist locally will be deleted remote.
The problem (bug) is that I did *not* delete any files locally. On the first pull some files were not downloaded to my local Google Drive mirror. On the first pull *all* the files (that don't have clashing filenames) should be downloaded, so something goes wrong.

I understand that when a file and a directory have the same name, then my filesystem cannot handle that. But it's not the case that the files that were not downloaded have the same path name as a directory.

I find it scary to skip the checksum-ing by using flag --ignore-checksum. I don't want to have to corrupt (lose) my content...

> Pulling specific files and folders instead of the whole thing. 

is a workaround that works for now.
But I would like to be able to do a full sync (push / pull), instead of always manually keeping everything exactly in sync.

Do you know what might have caused the problem that not all content was download to my local mirror on the first pull?
Do you have a solution for this problem?
Do you need any more information?

**Comment by barteend** [comment]
I tried to fix the problem by reinstalling drive....  ....but now the init doesn't work anymore.

What I did was:
* Uninstall drive
* Renamed my gdrive directory
* Restart my computer (just to be on the safe side)
* Install drive
* Go to https://drive.google.com -> Settings -> Manage apps, and disconnected Google Drive (removed the permissions of the old gdrive installation)
* In the console of my local computer:
   * `cd ~/Boo`
    * `mkdir gdrive`
    * `unbuffer drive init ~/Boo/gdrive 2>&1 | tee -a console-output.txt`

When I got on the point 'Paste the authorization code: ', I pasted the authorization code. After 5 minutes of waiting still nothing happened.
I pressed Ctrl+C and tried again.
Same thing...

So now I can't even initialize drive. So can't even use the workaround of "pulling specific files and folders instead of the whole thing" anymore.... :-(

If you know what I can do to use gdrive normally, please let me know

**Comment by odeke-em** [comment]
Hey @barteend I can't reply much right now but will do so when I get back home in about 8 hours.
Instead of the `unbuffer drive init ~/Boo/gdrive 2>&1 | tee -a console-output.txt` how about you just run
`drive init ~/Boo/gdrive`? This could just be due to the fact that you disabled input buffering using `unbuffer` yet Stdin is buffered.

**Comment by barteend** [comment]
Thank you @odeke-em !!!! :-) :+1: 
Leaving the unbuffer command away did the trick :-)

Now I'm running a 'drive pull 2>&1 | tee -a console-output.txt'. I hope it will download al my content this time...

**Comment by barteend** [comment]
Oh, using `drive pull 2>&1 | tee -a console-output.txt` is not such a good idea....

The file grows very fast, because every change in the progress bar is stored:
```
0 / 6458699279 [-----------------------------------------------------------------------------------------] 0.00 % 
19789 / 6458699279 [>--------------------------------------------------------------------------] 0.00 % 108h53m29s
54773 / 6458699279 [>----------------------------------------------------------------------------] 0.00 % 59h0m57s
915019 / 6458699279 [>----------------------------------------------------------------------------] 0.01 % 5h53m7s
```

I just post it so that other users who read this know that they might want to look for another way to get all the output from dive commands. I.e. increase the scrollback size of their console.

**Comment by barteend** [comment]
After the `drive pull` command completed I immediately executed another `drive pull` command, just to check if drive would find any missing files that weren't downloaded the first time.
I would expect that drive would download nothing, because everything should be downloaded on the first drive pull command.

Unfortunately the second execution of the `drive pull` command found a lot of files to add and even files to modify   ...and also files to delete. The files that were listed to be deleted are also listed to be added.
A short summary:
```
Addition count 432 src: 437.36MB
Modification count 31 src: 471.82KB dest: 85.67MB
Deletion count 15 dest: 564.87KB
```

I've checked for some of the files if they've got a clashing file name. For the files that I've checked this is not the case. Since it are hundreds of files and folders, I can't check them all manually.
Does drive provide a command which shows me all the conflicting filenames in Google Drive?

I can replicate the bug of my first time install of drive....

Do you know what might have caused the problem that not all content was download to my local mirror on the first pull?
Do you have a solution for this problem?
Do you need any more information?

**Comment by barteend** [comment]
Some additional info:

(Of course) a `drive push` command also shows a long list of changes. Here's the summary:

```
Deletion count 423 dest: 438.81MB
Modification count 22 src: 474.77KB dest: 471.82KB
Addition count 16 src: 2.00MB
```

**Comment by odeke-em** [comment]
Thank you for that. Soon, I'll add in a change that helps track clashing names then you won't have to go through this kind of trouble of debugging for days, this is the first time someone has reported this and I'd really like to make your experience with drive as pleasant as possible. My apologies if this is tripping you out. However, I'd encourage you maybe to do things directory centric e.g `cd this_directory`, `drive pull that_directory that_other_directory` otherwise there is way too much noise to try to narrow down what the issue is.
Please let me know if this makes sense.

**Comment by barteend** [comment]
Hi @odeke-em 

I appreciate that you really like to make everyone's experience with drive as pleasant as possible. With your very fast replies and good suggestions, you give me a *pleasant experience* even though drive doesn't work the way I want it to work :-) :+1: 

Could you post a message in this thread as soon as the clashing names list functionality is released? Then I know when I can check if I've got clashing filenames.

Personally I don't think my problem is a clashing filenames problem. Because a lot of files that I've checked don't have clashing file names....

Do you have any other idea what might have caused the problem that I've described?

P.S. the  `cd this_directory`, `drive pull that_directory that_other_directory` makes sense and works as a temporary workaround.

**Comment by odeke-em** [comment]
Hello @barteend,
 Thanks for understanding. For sure I will post a message informing you.
See the reason why I recommended `cd this_directory, drive pull that_directory that_other_directory` is not necessarily as a work around but there is a lot of uncertainty from your earlier pulls and pushes, I can't tell what completed what didn't. Posting the byte counts doesn't help, plus your logging method might have made you lose a bunch of error warnings that's why I recommended doing it in incremental small parts as well as using the vanilla commands `drive ...`. If however you would like for me to fully debug this, a manifest screenshot or console output would be great as lately you'll see a lot of this diagnosis was guess work since it seemed to be that drive was functioning as expected. Also you didn't mention what version of drive you were using `drive version`. If the output is sensitive, feel free to email it to me :)
Thank you.

**Comment by barteend** [comment]
Let's wait for the new version with clashing filenames listing functionality before starting to fully debug this... That might save us valuable time.

Thanks again!

**Comment by odeke-em** [comment]
Sounds like a plan, you can track its progress here https://github.com/odeke-em/drive/pull/166. It is mostly done. I was hoping to finish it tonight but I have an event this evening. However, I'll try to jump on it when I get back.

**Comment by odeke-em** [comment]
Please get the latest, clashing file detection now in.

**Comment by barteend** [comment]
I've got Kubuntu Trusty 14.04 LTS. So I've got to wait till the new version of drive is available as a Kubuntu 14.04 package....
